clojure-mode.el

```diff
@@ -979,18 +979,31 @@ nil."
 
 
 
-(defun clojure-expected-ns ()
-  "Return the namespace name that the file should have."
-  (let* ((project-dir (file-truename
-                       (or (locate-dominating-file default-directory
-                                                   "project.clj")
-                           (locate-dominating-file default-directory
-                                                   "build.boot"))))
-         (relative (substring (file-truename (buffer-file-name))
-                              (length project-dir)
-                              (- (length (file-name-extension (buffer-file-name) t))))))
-    (replace-regexp-in-string
-     "_" "-" (mapconcat 'identity (cdr (split-string relative "/")) "."))))
+(defun clojure-project-dir ()
```

Comment:
I mentioned this mostly because if we have the same function here, there'll be little point in fixing the problem in cider. Anyways, that's not terribly important right now.